File: androidApp/src/main/java/com/bracketcove/android/dashboards/passenger/PassengerDashboardUiState.kt
package com.bracketcove.android.dashboards.passenger

sealed interface PassengerDashboardUiState {
    object RideInactive: PassengerDashboardUiState
    data class SearchingForDriver(
        val passengerLat: Double,
        val passengerLon: Double,
        val destinationAddress: String
    ): PassengerDashboardUiState
    data class PassengerPickUp(
        val passengerLat: Double,
        val passengerLon: Double,
        val driverLat: Double,
        val driverLon: Double,
        val destinationLat: Double,
        val destinationLon: Double,
        val destinationAddress: String,
        val driverName: String,
        val driverAvatar: String
    ): PassengerDashboardUiState
    data class EnRoute(
        val passengerLat: Double,
        val passengerLon: Double,
        val destinationLat: Double,
        val destinationLon: Double,
        val destinationAddress: String,
        val driverName: String,
        val driverAvatar: String
    ): PassengerDashboardUiState

    data class Arrived(
        val passengerLat: Double,
        val passengerLon: Double,
        val destinationLat: Double,
        val destinationLon: Double,
        val destinationAddress: String,
        val driverName: String,
        val driverAvatar: String
    ): PassengerDashboardUiState

    //Signals something unexpected has happened
    object Error: PassengerDashboardUiState
    object Loading: PassengerDashboardUiState
}

File: iosApp/iosApp/ui/passengerdashboard/PassengerDashboardUiState.swift
//
//  PassengerDashboardUiState.swift
//  iosApp
//
//  Created by Ryan Kay on 2023-02-23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation

enum PassengerDashboardUiState {
    case rideInactive
    case loading
    case error
    
    struct SearchingForDriver {
        let passengerLat: Double
        let passengerLon: Double
        let destinationAddress: String
    }
    
    struct PassengerPickUp {
        let passengerLat: Double
        let passengerLon: Double
        let destinationAddress: String
        let destinationLat: Double
        let destinationLon: Double
        let driverName: String
        let driverAvatar: String
        let driverLat: Double
        let driverLon: Double
        let totalMessages: Int
    }
    
    struct EnRoute {
        let passengerLat: Double
        let passengerLon: Double
        let destinationAddress: String
        let destinationLat: Double
        let destinationLon: Double
        let driverName: String
        let driverAvatar: String
        let driverLat: Double
        let driverLon: Double
        let totalMessages: Int
    }
    
    struct Arrived {
        let passengerLat: Double
        let passengerLon: Double
        let destinationAddress: String
        let destinationLat: Double
        let destinationLon: Double
        let driverName: String
        let driverAvatar: String
        let totalMessages: Int
    }
}
